scripts/verifyZkSyncDeployments.ts

```diff
@@ -0,0 +1,161 @@
+// zksync block explorer does not support reading and writing to implementation contracts via a proxy
+// contract yet, so verification of the contract configuration/state must be done programatically.
+// To verify the zkSync SpokePool:
+//   yarn hardhat run --network zksync ./scripts/verifyZkSyncDeployments.ts
+//
+// To verify the zkSync adapter on mainnet:
+//   yarn hardhat run --network mainnet ./scripts/verifyZkSyncDeployments.ts
+
+import { assert } from "console";
+import { promises as fs } from "fs";
+import { ContractFactory } from "ethers";
+import { Provider as zkProvider, utils as zkUtils } from "zksync-web3";
+import { ethers, toBN, findArtifactFromPath, SignerWithAddress } from "../utils/utils";
+
+const textPadding = 30;
+
+/**
+ * l2RefundAddress and spokePoolDeploymentBlocks can change per deployment.
+ */
+const l2RefundAddress = "0x428AB2BA90Eba0a4Be7aF34C9Ac451ab061AC010";
+
+const tokens: { [chainId: number]: { symbol: string; address: string }[] } = {
+  1: [{ symbol: "WETH", address: "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2" }],
+  324: [{ symbol: "WETH", address: "0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91" }],
+};
+
+async function verifyAdapter(
+  chainId: number,
+  signer: SignerWithAddress,
+  deployments: Record<string, any>
+): Promise<void> {
+  const contractName = "ZkSync_Adapter";
+  assert([1, 5].includes(chainId), `Unexpected chain ID (${chainId})`);
```

Comment:
Will we ever want more than just 1,5? We should probably place this in a top level variable